File: core/src/test/java/com.memoizlabs.kotlin_playground/currency/GBPExchangeRateBuilder.java
package com.memoizlabs.kotlin_playground.currency;

public class GBPExchangeRateBuilder {

    private Money toUSD = new Money(0);
    private Money toEUR = new Money(0);

    public static GBPExchangeRateBuilder aGBPExchangeRate() {
        return new GBPExchangeRateBuilder();
    }

    public GBPExchangeRateBuilder withUsd(double usd) {
        toUSD = new Money(usd);
        return this;
    }

    public GBPExchangeRateBuilder withEur(double eur) {
        toEUR = new Money(eur);
        return this;
    }

    public GBPExchangeRate build() {
        return new GBPExchangeRate(toUSD, toEUR);
    }
}

File: core/src/test/java/com.memoizlabs.kotlin_playground/currency/GBPExchangeRateBuilder.kt
package com.memoizlabs.kotlin_playground.currency

class GBPExchangeRateBuilder {

    private var toUSD = Money(0.0)
    private var toEUR = Money(0.0)

    fun withUsd(usd: Double): GBPExchangeRateBuilder {
        toUSD = Money(usd)
        return this
    }

    fun withEur(eur: Double): GBPExchangeRateBuilder {
        toEUR = Money(eur)
        return this
    }

    fun build(): GBPExchangeRate {
        return GBPExchangeRate(toUSD, toEUR)
    }

    companion object {

        fun aGBPExchangeRate(): GBPExchangeRateBuilder {
            return GBPExchangeRateBuilder()
        }
    }
}
